packages/material-ui/src/Slider/Slider.test.js

```diff
@@ -701,9 +872,11 @@ describe('<Slider />', () => {
 
       expect(() => {
         setProps({ value: undefined });
-      }).toErrorDev(
+      }).toErrorDev([
+        // TODO Not sure if we should try and silence this warning or if it is ok to keep
```

Comment:
I would be leaning toward yes, as it creates noise compared to the second warning.

Reply:
I can silence the error if the user initializes the component without a value with something like 

```
const [valueDerived, setValueState] = useControlled({
    controlled: valueProp,
    default: defaultValue ?? min,
    name: 'Slider',
  });
```

Where if they don't provide a `value` or a `defaultValue` it will default to `min` which is in turn defaulted to `0` which is close to what happens without this code (since visually NaN shows the slider at 0). 

But it is a bit more complicated with the test where it takes a controlled slider and converts it to un-controlled since `useControlled` ignores the default if the input was ever controlled. 

The NaN is introduced in the `clamp` method so I think the cleanest option is to change the clamp method to:

```
function clamp(value, min, max) {
  if (value == null) {
    return min;
  }
  return Math.min(Math.max(min, value), max);
}
```

I think these two changes will be the cleanest ways to silence the error are provide the sanest fallbacks.

Reply:
OK, looks good.